packages/cli/templates/default-plugin/package.json.hbs

```diff
@@ -32,7 +32,8 @@
     "@material-ui/lab": "4.0.0-alpha.45",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
-    "react-use": "^17.2.4"
+    "react-use": "^17.2.4",
+    "@types/react": "^16.9"
```

Comment:
I think it's best to move this into `devDeps` and if people export things that rely on exporting the react types then the script will tell them to move it, rather than it being installed always.

Reply:
@iamEAP had an idea for why it may actually be needed in the dependencies, which is why I included it there. Could you perhaps clarify your reasoning, Eric?

Reply:
I was thinking of [this](https://github.com/facebook/create-react-app/issues/6180#issuecomment-453640473), where...  Because Backstage frontend plugins end up built/bundled anyway, the distinction between devDeps vs. deps is fairly minor?

Reply:
Yep agreed. We could encode this in an ADR even. But I wonder if our tooling will actually complain about this anyway - you know, the build check that tells you to move things to/from dev deps

Reply:
Ah I was not in the thread I thought I was, haha. The check had already been mentooned

Reply:
Come to think of it, does this translate as well to libraries? I mean sure, a web application gets built in the end into a static bundle, but what about the `-react` packages and core etc? Does it matter there, where the dependencies lie?

Reply:
Kind of ironic that the CI fails on a missing @types/react dependency in one of the plugins.

To me it seems like this dependency surprisingly isn't needed for all the frontend plugins. Unless you think there's any more to discuss, feel free to close this @freben or @benjdlambert.

Reply:
I think if you move it to `devDeps` there's a chance that error message should go away. I don't think it enforces `devDeps`

Reply:
It's not required but I think it's good to have and it's a little clearer that you have to move it from devDeps to deps given the error message you get when you export a react component because your rexporting the types.

Reply:
Alright! I have added it to devDeps instead now :tada: